### PR TITLE
Make `--output` flag support all types of paths

### LINKED ISSUE #9090: Improve --output flag for the commands to support all types of paths
### Description

Create a generic utility for handling `--output` flag in Lisk commander. There are a few commands that use --output flag to output results to a specific directory along with filename or without it.

- By default we write to current directory `cwd`
- If a path is given in any form, with `~` or relative or absolute path needs to be resolved.
- If the path is given along with filename then choose that custom file name otherwise use a default one, preferably namespace `passphrase:create` to `passphrase.json` or `keys:create` to `keys.json` 

### Acceptance Criteria

- All commands using `--output` flag should have same behaviour
- Should have all unit tests

### PR DESCRIPTION
**Pull request by mosmartin**
### What was the problem?

This PR resolves #9090

### How was it solved?

Created the `handleOutputFlag` utility function.
With this utiliy, it's now possible to generate output files with following commands

1. `./bin/run generator export`  will generate a file with the default namespace name in the `cwd`.
2. `./bin/run generator export --output .`  will generate a file with the default namespace name in the `cwd`.
3. `./bin/run generator export --output ./` will generate a file with the default namespace name in the `cwd`.
4. `./bin/run generator export --output test.json` will generate the file with the provided filename in the `cwd`.
5. `./bin/run generator export --output ~` will generate a file with the default namespace name in the user home dir.
6. `./bin/run generator export --output ~/Desktop/` will generate a file with the default namespace name in the user desktop.
7. `./bin/run generator export --output exports` will generate a file with the default namespace name in the defined dir (`exports`) in `cwd`
8. `./bin/run generator export --output /Users/johndoe/Public/test.json` will generate the `test.json` file in the defined path

### How was it tested?

Updated and added new tests

### CODE REVIEW

File: commander/test/bootstrapping/commands/keys/create.spec.ts [hunk 1]
```diff
@@ -20,7 +20,8 @@ import * as readerUtils from '../../../../src/utils/reader';
 import { CreateCommand } from '../../../../src/bootstrapping/commands/keys/create';
 import { getConfig } from '../../../helpers/config';
 import { Awaited } from '../../../types';
-import { OWNER_READ_WRITE } from '../../../../src/constants';
+// import { OWNER_READ_WRITE } from '../../../../src/constants';
```
[the remark this answers is not in the document]
Comment (mosmartin) [reply commander/test/bootstrapping/commands/keys/create.spec.ts:23]: Ahh yes, need to clean up the comments 🙂

File: commander/src/bootstrapping/commands/passphrase/encrypt.ts [hunk 2]
```diff
@@ -54,17 +55,13 @@ export class EncryptCommand extends Command {
...
-			fs.ensureDirSync(dir);
-		}
-
 		const passphrase = passphraseSource ?? (await getPassphraseFromPrompt('passphrase', true));
 		const password = passwordSource ?? (await getPasswordFromPrompt('password', true));
 		const result = await encryptPassphrase(passphrase, password, outputPublicKey);
 
 		if (output) {
-			fs.writeJSONSync(output, result, { spaces: ' ', mode: OWNER_READ_WRITE });
+			const res = await handleOutputFlag(output, result, 'passphrase', 'keys.json');
```
Comment (ishantiw) [inline commander/src/bootstrapping/commands/passphrase/encrypt.ts:63]:
```suggestion
			const res = await handleOutputFlag(output, result, 'passphrase', 'passphrase');
```

File: commander/src/bootstrapping/commands/config/create.ts [hunk 3]
```diff
@@ -81,17 +81,14 @@ export class CreateCommand extends Command {
...
+				this.log(res);
 			}
 		} else {
 			fs.mkdirSync(configPath, { recursive: true });
-			fs.writeJSONSync(resolve(configPath, 'config.json'), defaultConfig, {
-				spaces: '\t',
-				mode: OWNER_READ_WRITE,
-			});
+
+			const res = await handleOutputFlag(configPath, defaultConfig, 'config', 'config.json');
```
Comment (ishantiw) [inline commander/src/bootstrapping/commands/config/create.ts:90]:
```suggestion
			const res = await handleOutputFlag(configPath, defaultConfig, 'config', 'config');
```

sometimes we are passing with extension and sometimes we don't. Maybe to use one standard

File: commander/src/bootstrapping/commands/keys/export.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (ishantiw) [inline commander/src/bootstrapping/commands/keys/export.ts:1]:
```suggestion
```

File: commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts [hunk 5]
```diff
@@ -92,59 +92,75 @@ describe('passphrase:encrypt', () => {
...
 			);
 		});
 	});
 
 	describe('passphrase:encrypt --passphrase="enemy pill squeeze gold spoil aisle awake thumb congress false box wagon"  --output=/mypath/keys.json', () => {
 		it('should encrypt passphrase from passphrase flag and stdout password', async () => {
+			jest
+				.spyOn(outputUtils, 'handleOutputFlag')
+				.mockImplementation(async () =>
+					Promise.resolve('Successfully written data to /my/path/keys.json'),
```
Comment (ishantiw) [inline commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts:127]:
```suggestion
					Promise.resolve('Successfully written data to /my/path/passphrase.json'),
```

File: commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts [hunk 6]
```diff
@@ -92,59 +92,75 @@ describe('passphrase:encrypt', () => {
...
 				'/mypath/keys.json',
 				{
 					encryptedPassphrase: encryptedPassphraseObject,
 				},
-				{
-					spaces: ' ',
-					mode: OWNER_READ_WRITE,
-				},
+				'passphrase',
+				'keys.json',
```
Comment (ishantiw) [inline commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts:151]:
```suggestion
				'passphrase.json',
```

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: commander/src/bootstrapping/commands/config/create.ts (hunk 3 above)
Comment (mosmartin) [reply commander/src/bootstrapping/commands/config/create.ts:90]: The filename is optional. If not present then the namespace is used. But yes, we can omit the filename unless necessary.

**Review by ishantiw — APPROVED** [review]

**Pull request merged**